File: Client/Roll10.Domain.Tests/HelperTests.fs
module Roll10.Domain.Tests.HelperTests

open System.Text.RegularExpressions
open Roll10.Domain.Helpers
open NUnit.Framework

[<SetUp>]
let Setup () =
    ()

[<Test>]
let GenerateIdTest () =
    let id = generateId
    let regexMatch = Regex("[a-z]+").IsMatch(id)
    Assert.AreEqual(15, id.Length)
    Assert.True(regexMatch)
    
[<Test>] 
let RemoveEmptyStringsFromList () =
    let list = ["one";"";"two"]
    let parsed = list |> removeEmptyStrings
    Assert.AreEqual(["one";"two"], parsed)
    

File: Client/Roll10.Domain.Tests/HelperTests.cs
using System.Text.RegularExpressions;
using Roll10.Domain.Services;

namespace Roll10.Domain.Tests;

public class HelperTests
{
    [SetUp]
    public void Setup()
    {
    }

    [Test]
    public void CanCreateId()
    {
        for (var x = 0; x < 100; x++)
        {
            var id = Helpers.GenerateId();
            Assert.That(new Regex("[a-z0-9]{15}").IsMatch(id), id);
        }
    }
}